local-storage-service/server/src/main/java/org/verapdf/localstorageservice/server/controller/FileController.java

```diff
@@ -0,0 +1,43 @@
+package org.verapdf.localstorageservice.server.controller;
+
+import org.springframework.core.io.Resource;
+import org.springframework.data.util.Pair;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.verapdf.localstorageservice.model.dto.StoredFileDTO;
+import org.verapdf.error.exception.NotFoundException;
+import org.verapdf.error.exception.VeraPDFBackendException;
+import org.verapdf.localstorageservice.server.service.StoredFileService;
+
+import java.io.FileNotFoundException;
+import java.util.UUID;
+
+@RestController
+public class FileController {
+
+	private final StoredFileService storedFileService;
+
+	public FileController(StoredFileService storedFileService) {
+		this.storedFileService = storedFileService;
+	}
+
+	@PostMapping()
+	public StoredFileDTO uploadFile(@RequestPart("file") MultipartFile file,
+	                                @RequestParam("checksum") String checksum) throws VeraPDFBackendException {
```

Comment:
1. I expect that we will upload files one by one. That way it will be easier to Retry single file upload if it fails for some reason.
2. Indeed, I expect it to be a request part as I described in API doc
3. Name is up to you, just make sure to update documentation accordingly

Reply:
@yuliya-ivaniukovich So can we return to Body as a file (@RequestBody) and Content-MD5 header in the request for checksum? For 1 file per upload I do not think that we need multipart form for that (i.e. request part)

Reply:
As discussed, unfortunately we can only send files as `multipart/form-data` from the browser so let's keep this format. You're free to add alternative endpoint for internal use though (e.g. to save validation report from worker).